app/src/main/java/com/github/polybooks/core/Interests.kt

```diff
@@ -0,0 +1,26 @@
+package com.github.polybooks.core
+
+/**
+ * An Interest is something that can make a Book interesting for a user.
+ * Interests can be Courses, Fields or Semesters.
+ * */
+sealed class Interest
+
+/**
+ * A course that a student can follow (e.g. COM-101)
+ * @property courseName The identifier name of that course
+ * */
+data class Course(val courseName : String) : Interest()
+
+/**
+ * A field of study.
+ * @property fieldName The identifier name of that field
+ * */
+data class Field(val fieldName : String) : Interest()
+
+/**
+ * Represents the pair section-semester a student can be in (e.g. IC BA6)
+ * @property section The identifier string of that section.
+ * @property semester The identifier string of that semester.
+ * */
+data class Semester(val section : String, val semester : String) : Interest()
```

Comment:
I disagree with your Semester vision:
- What is the difference betwen Field and Semester.section concretely ?
- From an UX point of view, when we'll offer filters to users, we don't want to have a massive dropdown with ~100 choices of IC BA1, IC BA2, ... It's much better to have 2 dropdowns with 10 choices. Selecting first IC, then BA6
- The fact that we have Semester.semester seems to me like a "code smell" that something is wrong with this data structure. At the bare minimum calling it Semester, but I'd say the whole structure is not needed.

Reply:
Field would be independent of the big sections of EPFL and could be more fine grained, like Quantum computing, Biochemistry. We could also not include a third type of interest.
From the point of view of UX, the structure of the Semester class does not prevent selecting first IC then BA3.
I agree about the semester.semester, I think it's mostly a problem of vocabulary, I'm open to propositions for another word to describe something like BA2,BA3,MA2,...